type-generator/build.sbt

```diff
@@ -16,5 +16,10 @@ lazy val root = (project in file("."))
       Tests.Argument(
         "-u", "target/test-reports", // TODO(MARATHON-8215): Remove this line
         "-o", "-eDFG",
-        "-y", "org.scalatest.WordSpec"))
+        "-y", "org.scalatest.WordSpec")),
+    scalacOptions in Compile ++= Seq(
+      "-encoding", "UTF-8",
+      "-target:jvm-1.8",
+      "-release", "8"
```

Comment:
could we maybe mention this issue in the comment? https://github.com/scala/bug/issues/11125 so we know the context of this change later... (without traversing git history)

Reply:
Done